google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITQueryAggregationsTest.java

```diff
@@ -1,5 +1,750 @@
+    assertThat(snapshot.get(sum("pages"))).isEqualTo(200);
+    assertThat(snapshot.get(average("pages"))).isEqualTo(100);
+    assertThat(snapshot.get(AggregateField.count())).isEqualTo(2);
+  }
+
+  @Test
+  public void performsAggregationsOnNestedMapValues() throws Exception {
+    Map<String, Map<String, Object>> testDocs =
+        map(
+            "a",
+            map(
+                "author",
+                "authorA",
+                "title",
+                "titleA",
+                "metadata",
+                map("pages", 100, "rating", map("critic", 2, "user", 5))),
+            "b",
+            map(
+                "author",
+                "authorB",
+                "title",
+                "titleB",
+                "metadata",
+                map("pages", 50, "rating", map("critic", 4, "user", 4))));
+    CollectionReference collection = testCollectionWithDocs(testDocs);
+    AggregateQuerySnapshot snapshot =
+        collection
+            .aggregate(
+                sum("metadata.pages"),
+                average("metadata.pages"),
+                average("metadata.rating.critic"),
+                sum("metadata.rating.user"),
+                AggregateField.count())
+            .get()
+            .get();
+    assertThat(snapshot.get(sum("metadata.pages"))).isEqualTo(150);
+    assertThat(snapshot.get(average("metadata.pages"))).isEqualTo(75);
+    assertThat(snapshot.get(average("metadata.rating.critic"))).isEqualTo(3);
+    assertThat(snapshot.get(sum("metadata.rating.user"))).isEqualTo(9);
+    assertThat(snapshot.get(AggregateField.count())).isEqualTo(2);
+  }
+
+  @Test
+  public void performsSumThatResultsInFloat() throws Exception {
+    Map<String, Map<String, Object>> testDocs =
+        map(
+            "a", map("author", "authorA", "title", "titleA", "rating", 5),
+            "b", map("author", "authorB", "title", "titleB", "rating", 4.5),
+            "c", map("author", "authorC", "title", "titleC", "rating", 3));
+    CollectionReference collection = testCollectionWithDocs(testDocs);
+    AggregateQuerySnapshot snapshot = collection.aggregate(sum("rating")).get().get();
+    assertThat(snapshot.get(sum("rating"))).isEqualTo(12.5);
+  }
+
+  @Test
+  public void performsSumOfIntsAndFloatsThatResultsInInt() throws Exception {
+    Map<String, Map<String, Object>> testDocs =
+        map(
+            "a", map("author", "authorA", "title", "titleA", "rating", 5),
+            "b", map("author", "authorB", "title", "titleB", "rating", 4.5),
+            "c", map("author", "authorC", "title", "titleC", "rating", 3.5));
+    CollectionReference collection = testCollectionWithDocs(testDocs);
+    AggregateQuerySnapshot snapshot = collection.aggregate(sum("rating")).get().get();
+    assertThat(snapshot.get(sum("rating"))).isEqualTo(13);
+  }
+
+  // TODO: Does Integer.MAX_VALUE serve the purpose of this test?
+  @Test
+  public void performsSumThatOverflowsMaxInt() throws Exception {
```

Comment:
This is an error in Web SDK and here in Java. The intent is to add two integers and overflow the 64-bit integer representation on the server, causing the server to return a double. Long.MAX_VALUE is probably a better choice for java-firestore. I'll have to determine how this can be tested in Web.

Reply:
Ah! Cool. Fixed.